src/pcm/SoxrResampler.cxx

```diff
@@ -80,20 +90,118 @@ soxr_parse_quality(const char *quality) noexcept
 	return SOXR_INVALID_RECIPE;
 }
 
+static unsigned
+SoxrParsePrecision(unsigned value) {
+	switch (value) {
+		case 16:
+		case 20:
+		case 24:
+		case 28:
+		case 32:
+			break;
+		default:
+			throw FormatInvalidArgument(
+				"soxr converter invalid precision : %d [16|20|24|28|32]", value);
+			break;
+	}
+	return value;
+}
+
+static double
+SoxrParsePhaseRespons(unsigned value) {
+	if (value > 100) {
+		throw FormatInvalidArgument(
+			"soxr converter invalid phase_respons : %d (0-100)", value);
+	}
+
+	return double(value);
+}
+
+static double
+SoxrParsePassbandEnd(const char *svalue) {
+	char *endptr;
+	double value = strtod(svalue, &endptr);
+	if (svalue == endptr) {
```

Comment:
This doesn't check for garbage after the parsed number

Reply:
When no number could be parsed the a expection is rased and if the number isn't within the expected range also.
Can you clarify ? You mean garbage as like a string with "1.23 ABC" where only the first part is parsed ?

Reply:
> You mean garbage as like a string with "1.23 ABC" where only the first part is parsed ?

Yes. Such a string should raise an error, because MPD should not silently ignore such garbage.

Usually, after `strto*()`, you check for errors using `endptr == s || *endptr != 0`. This checks whether anything was parsed at all, and whether everything was parsed.